x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/async/QlStatusResponse.java

```diff
@@ -4,40 +4,48 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-package org.elasticsearch.xpack.eql.action;
+package org.elasticsearch.xpack.ql.async;
 
 import org.elasticsearch.ExceptionsHelper;
 import org.elasticsearch.action.ActionResponse;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.StatusToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.xpack.core.search.action.SearchStatusResponse;
-import org.elasticsearch.xpack.eql.async.StoredAsyncResponse;
 
 import java.io.IOException;
 import java.util.Objects;
 
 import static org.elasticsearch.rest.RestStatus.OK;
 
 /**
- * A response for eql search status request
+ * A response for *QL search status request
  */
-public class EqlStatusResponse extends ActionResponse implements SearchStatusResponse, StatusToXContentObject {
+public class QlStatusResponse extends ActionResponse implements SearchStatusResponse, StatusToXContentObject {
     private final String id;
     private final boolean isRunning;
     private final boolean isPartial;
     private final Long startTimeMillis;
     private final long expirationTimeMillis;
     private final RestStatus completionStatus;
 
-    public EqlStatusResponse(String id,
-            boolean isRunning,
-            boolean isPartial,
-            Long startTimeMillis,
-            long expirationTimeMillis,
-            RestStatus completionStatus) {
+    public interface AsyncStatus {
```

Comment:
Imho, I would extract this to its own file.

Reply:
I find the interface tightly bound to the outer class (i.e. no independent applicability), that's why I was also using it as a prefix wherever the interface is used. But if you think it'd be better practice to extract it, I can still rename it and do it.

(Thanks for this long-PR review!)

Reply:
I'll leave it up to you, not insisting, it's just that from the name `AsyncStatus` it seems like a generic iface and could also belong to a separate file.